### ISSUE TITLE
Something wrong with [char-class]{quantity}

### ISSUE THREAD

**Issue by AlphaKoder** [asciimoo/exrex#27]
I am seeing different counts than expected when using braced fixed-quantities after character class expressions.  There is a regex for the numeric range 0 to 65535, whose count should be 65536:

```
>>> exrex.count('^([0-9]|[1-9][0-9]|[1-9][0-9][0-9]|[1-9][0-9][0-9][0-9]|[1-5][0-9][0-9][0-9][0-9]|6[0-4][0-9][0-9][0-9]|65[0-4][0-9][0-9]|655[0-2][0-9]|6553[0-5])$')                
65536
```

While count() works great in the above case, it does not when I condense the regex using braced fixed-quantities:
```
>>> exrex.count('^([0-9]|[1-9][0-9]|[1-9][0-9]{2}|[1-9][0-9]{3}]|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$')
12369
```

To make it more interesting, the braced fixed quantities work again it they are grouped using parentheses with the expression they are repeating:

```
>>> exrex.count('^([0-9]|[1-9][0-9]|[1-9]([0-9]{2})|[1-9]([0-9]{3})]|[1-5]([0-9]{4})|6[0-4]([0-9]{3})|65[0-4]([0-9]{2})|655[0-2][0-9]|6553[0-5])$')
65536
```

So, there appears to be a bug somewhere in what gets generated when the regex includes quantities for character classes and using grouping to ensure the quantity applies to just the preceding character class appears to address it.  Thanks!

**Comment by asciimoo** [comment]
Related to #26

**Comment by AlphaKoder** [comment]
In case this helps, the issue appears to be specifically related to count().  Getting the length of the list of items from generate() produces the correct count:

```
>>> len(list(exrex.generate('^([0-9]|[1-9][0-9]|[1-9][0-9]{2}|[1-9][0-9]{3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$')))
65536
```

**Comment by asciimoo** [comment]
@AlphaKoder thanks for the help. This is the shortest regex which can reproduce the bug: `\da{2}`. The problem is in this `if` branch: https://github.com/asciimoo/exrex/blob/master/exrex.py#L197 . I'll try to fix it soon

**Comment by AlphaKoder** [comment]
Thank you.

**Comment by asciimoo** [comment]
fixed in 5f615d0
